website/src/theme/DocItem/index.js

```diff
@@ -49,6 +50,10 @@ export default function DocItem(props) {
   // If term has cta property set, show that cta
   const termCTA = frontMatter?.cta && frontMatter.cta
 
+  // dbt Custom
+  // If the page has a searchWeight value, apply that value
+  const searchWeight = frontMatter?.searchWeight && frontMatter.searchWeight
```

Comment:
For this I followed the pattern we were using for `termCTA` and in this case I believe the && is being used as a short circuit operator between the two statements (e.g. `const searchWeight = A && B` - B is only evaluated if A is truthy).

Looking at it now since we are using optional chaining with `frontMatter?.searchWeight`, I don't think it's necessary to do a short circuit eval. @JKarlavige thoughts on just using `const searchWeight = frontMatter?.searchWeight` here?

Reply:
Yeah that should work fine, I believe the shorter has the same outcome as the longer statement. If `frontMatter` or `frontMatter.searchWidget` is not set, `const searchWeight` will = `undefined`.

Reply:
Okay -- so I think I understand. Let me read up a little bit more on what `?` does, because I've seen you folks using it a lot and I assumed it was a TS thing 😆 .

Reply:
@lfracalossiFA The way i look at it is it being a shorthanded `if` statement for checking nested properties in objects. For example, the statement is shorthand for:
```
if(frontMatter && frontMatter.searchWidget)
```

Doing `frontMatter?.searchWidget` says if `frontMatter` is not set, return false, otherwise proceed to check if `searchWidget` is set.